### DISCUSSION TITLE
brew outdated --greedy

### DISCUSSION THREAD

**Discussion by bevanjkay** [Homebrew/discussions#515] [Everyday usage]
I'm looking for a way to stop `brew outdated --greedy` from outputting casks that are listed as version `latest`.

I like to keep tabs on what applications are updating, so often upgrade them manually from this list, but after installing a few fonts, there are a lot of casks appearing with `(latest) != latest` that I would prefer not to see.

**Comment by carlocab** [comment]
Still thinking about it, but there's probably a way to parse the output of `brew outdated --greedy --json` in order to exclude the casks with version = `latest`.

It's probably not even hard, but my `jq`-fu is totally lacking

**Reply by bevanjkay** [reply]
@carlocab This is exactly what I needed! Thanks

`brew outdated --greedy --json | jq -r '.casks | map(select(.current_version != "latest")) | map("\(.name) (\(.current_version)) != \(.installed_versions)") | join("\n")'`

**Comment by gromgit** [comment]
A solution that only uses system utilities:
```
brew outdated --greedy --verbose | grep -v '(latest)' | cut -d' ' -f1
```

**Reply by carlocab** [reply]
Piping to `grep` would've been my preferred solution too, except I didn't know how to stop `brew` from eating the version information before piping the formula names out. Turns out it was `--verbose`.

**Reply by bevanjkay** [reply]
@gromgit Thanks. That's a much simpler solution.

**Reply by toobuntu** [reply]
One less process:
`brew outdated --greedy --verbose | awk '$2 !~ /(latest)/ {print $1}'`
